pandas/_libs/lib.pyx

```diff
@@ -2566,6 +2567,8 @@ def maybe_convert_objects(ndarray[object] objects,
         encountered, whether to convert and return an Boolean/IntegerArray.
     convert_non_numeric : bool, default False
         Whether to convert datetime, timedelta, period, interval types.
+    dtype_if_all_na : np.dtype, ExtensionDtype, or None, default None
+        Dtype to cast to if we have all-NA or all-None.
     dtype_if_all_nat : np.dtype, ExtensionDtype, or None, default None
```

Comment:
can this replace dtype_if_all_nat instead of adding a new arg?

Reply:
never mind, i now see that isnt viable.

Reply:
Yes, It seems not easy. the details are above. https://github.com/pandas-dev/pandas/pull/62352#discussion_r2360248114

There will be pros and cons, but I think the logic of `_cast_pointwise_result` is that the current version is clearer.